Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean

```diff
@@ -0,0 +1,685 @@
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac)
+    (hg : ContinuousOn g (spectrum R a) := by cfc_cont_tac) :
+    cfc a (fun x ↦ f x + g x) = cfc a f + cfc a g := by
+  by_cases ha : p a
+  · rw [cfc_apply a f, cfc_apply a g, ← map_add, cfc_apply a _]
+    congr
+  · simp [cfc_apply_of_not_predicate a ha]
+
+lemma cfc_smul {S : Type*} [SMul S R] [ContinuousConstSMul S R]
+    [SMulZeroClass S A] [IsScalarTower S R A] [IsScalarTower S R (R → R)]
+    (s : S) (f : R → R) (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    cfc a (fun x ↦ s • f x) = s • cfc a f := by
+  by_cases ha : p a
+  · rw [cfc_apply a f, cfc_apply a _]
+    simp_rw [← Pi.smul_def, ← smul_one_smul R s _]
+    rw [← map_smul]
+    congr
+  · simp [cfc_apply_of_not_predicate a ha]
+
+lemma cfc_const_mul (r : R) (f : R → R)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    cfc a (fun x ↦ r * f x) = r • cfc a f :=
+  cfc_smul a r f
+
+lemma cfc_star (f : R → R) : cfc a (fun x ↦ star (f x)) = star (cfc a f) := by
+  by_cases h : p a ∧ ContinuousOn f (spectrum R a)
+  · obtain ⟨ha, hf⟩ := h
+    rw [cfc_apply a f, ← map_star, cfc_apply a _]
+    congr
+  · obtain (ha | hf) := not_and_or.mp h
+    · simp [cfc_apply_of_not_predicate a ha]
+    · rw [cfc_apply_of_not_continuousOn a hf, cfc_apply_of_not_continuousOn, star_zero]
+      exact fun hf_star ↦ hf <| by simpa using hf_star.star
+
+lemma cfc_pow_id (n : ℕ) (ha : p a := by cfc_tac) : cfc a (· ^ n : R → R) = a ^ n := by
+  rw [cfc_pow a _, cfc_id' R a]
+
+lemma cfc_smul_id {S : Type*} [SMul S R] [ContinuousConstSMul S R]
+    [SMulZeroClass S A] [IsScalarTower S R A] [IsScalarTower S R (R → R)]
+    (s : S) (ha : p a := by cfc_tac) : cfc a (s • · : R → R) = s • a := by
+  have := cfc_id R a ▸ cfc_smul a s id
+  exact this
+
+lemma cfc_const_mul_id (r : R) (ha : p a := by cfc_tac) : cfc a (r * ·) = r • a :=
+  cfc_smul_id a r
+
+lemma cfc_star_id (ha : p a := by cfc_tac) : cfc a (star · : R → R) = star a := by
+  nth_rw 2 [← cfcHom_map_id (show p a from ha) (R := R)]
+  rw [← map_star, cfc_apply a (star : R → R)]
+  congr
+
+section Polynomial
+open Polynomial
+
+lemma cfc_eval_X (ha : p a := by cfc_tac) :
+    cfc a (X : R[X]).eval = a := by
+  simpa using cfc_id R a
+
+lemma cfc_eval_C (r : R) (ha : p a := by cfc_tac) :
+    cfc a (C r).eval = algebraMap R A r := by
+  simp [cfc_const a r]
+
+lemma cfc_map_polynomial (q : R[X]) (f : R → R) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    cfc a (fun x ↦ q.eval (f x)) = aeval (cfc a f) q := by
+  induction q using Polynomial.induction_on with
+  | h_C r => simp [cfc_const a r]
+  | h_add q₁ q₂ hq₁ hq₂ =>
+    simp only [eval_add, map_add, ← hq₁, ← hq₂, cfc_add a (q₁.eval <| f ·) (q₂.eval <| f ·)]
+  | h_monomial n r _ =>
+    simp only [eval_mul, eval_C, eval_pow, eval_X, map_mul, aeval_C, map_pow, aeval_X]
+    rw [cfc_const_mul .., cfc_pow _ _ (n + 1),
+      ← smul_eq_mul, algebraMap_smul]
```

Comment:
If you add
```lean
open Polynomial in
lemma cfcHom_map_polynomial (q : R[X]) (f : C(spectrum R a, R)) :
    cfcHom ha (aeval f q) = aeval (cfcHom ha f) q :=
  .symm <| aeval_algHom_apply _ _ _
```
as well as
```lean
@[simp]
lemma ContinuousMap.aeval_apply {X R A : Type*} [CommSemiring R] [TopologicalSpace X]
    [TopologicalSpace A] [CommSemiring A] [Algebra R A] [TopologicalSemiring A]
    (f : C(X, A)) (q : R[X]) (x : X) :
    aeval f q x = aeval (f x) q := by
  rw [← coeFnAlgHom_apply R, ← aeval_algHom_apply]
  -- note : aeval_fn_apply should be generalized to algebras
  exact .symm <| aeval_algHom_apply (Pi.evalAlgHom R (fun _ ↦ A) x) _ _
```
you can do
```suggestion
  rw [cfc_apply a _, cfc_apply a _, ← cfcHom_map_polynomial]
  congr 1
  ext x
  simp
```

This can wait for another PR.

Reply:
Nice! I'll wait for another PR just because `ContinuousMap.aeval_apply` should go in a different file.